### PR TITLE
Remove sigs.k8s.io/cluster-api dependency

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

It removes the sigs.k8s.io/cluster-api dependency in order to allow us to update the controller-runtime version. This is not possible upstream, because they are [not maintaining the release-0.1 branch anymore](https://github.com/kubernetes-sigs/cluster-api/pull/1630#issuecomment-544991807)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

```release-note
none
```

### CODE REVIEW

File: pkg/admission/machinedeployments_validation.go [hunk 1]
```diff
@@ -17,15 +17,15 @@ limitations under the License.
 package admission
 
 import (
+	"github.com/kubermatic/machine-controller/pkg/apis/cluster/common"
```
Comment (xmudrii) [inline pkg/admission/machinedeployments_validation.go]: Imports in this file could eventually be organized better.

File: pkg/admission/machines.go [hunk 2]
```diff
@@ -22,12 +22,11 @@ import (
 
 	"golang.org/x/crypto/ssh"
 
+	clusterv1alpha1 "github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
```
Comment (xmudrii) [inline pkg/admission/machines.go]: Imports nit: this should eventually go with `machine-controller` imports

File: pkg/apis/cluster/v1alpha1/conversions/conversions.go [hunk 3]
```diff
@@ -22,18 +22,12 @@ import (
 
 	machinesv1alpha1 "github.com/kubermatic/machine-controller/pkg/machines/v1alpha1"
 
-	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	clusterv1alpha1 "github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
```
Comment (xmudrii) [inline pkg/apis/cluster/v1alpha1/conversions/conversions.go]: Imports nit: space not needed between those two imports.

File: pkg/apis/cluster/v1alpha1/conversions/conversions_test.go [hunk 4]
```diff
@@ -27,7 +27,7 @@ import (
 
 	machinesv1alpha1 "github.com/kubermatic/machine-controller/pkg/machines/v1alpha1"
 
-	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	clusterv1alpha1 "github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
```
Comment (xmudrii) [inline pkg/apis/cluster/v1alpha1/conversions/conversions_test.go]: Imports nit: space not needed between those two imports.

File: pkg/apis/cluster/v1alpha1/machine_types.go [hunk 5]
```diff
@@ -17,10 +17,10 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"github.com/kubermatic/machine-controller/pkg/apis/cluster/common"
```
Comment (xmudrii) [inline pkg/apis/cluster/v1alpha1/machine_types.go]: Imports nit: split those to two groups.

File: pkg/apis/cluster/v1alpha1/machinedeployment_types.go [hunk 6]
```diff
@@ -17,9 +17,9 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"github.com/kubermatic/machine-controller/pkg/apis/cluster/common"
```
Comment (xmudrii) [inline pkg/apis/cluster/v1alpha1/machinedeployment_types.go]: Imports nit: split those to two groups.

File: pkg/apis/cluster/v1alpha1/migrations/migrations.go [hunk 7]
```diff
@@ -30,6 +30,7 @@ import (
 	machinesv1alpha1 "github.com/kubermatic/machine-controller/pkg/machines/v1alpha1"
 	"github.com/kubermatic/machine-controller/pkg/providerconfig"
 
+	clusterv1alpha1 "github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
```
Comment (xmudrii) [inline pkg/apis/cluster/v1alpha1/migrations/migrations.go]: Imports nit: should eventually go in the group above.

File: pkg/apis/cluster/v1alpha1/zz_generated.deepcopy.go [hunk 8]
```diff
@@ -21,10 +21,10 @@ limitations under the License.
 package v1alpha1
 
 import (
+	common "github.com/kubermatic/machine-controller/pkg/apis/cluster/common"
```
Comment (xmudrii) [inline pkg/apis/cluster/v1alpha1/zz_generated.deepcopy.go]: Imports nit: split those to two groups.

File: pkg/controller/machineset/status.go [hunk 9]
```diff
@@ -132,3 +132,51 @@ func (c *ReconcileMachineSet) getMachineNode(machine *v1alpha1.Machine) (*corev1
...
+	for _, c := range node.Status.Conditions {
+		if c.Type == corev1.NodeReady {
+			return c.Status == corev1.ConditionTrue
+		}
+	}
+	return false
+}
+
+// getReadyCondition extracts the ready condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
```
Comment (xmudrii) [inline pkg/controller/machineset/status.go]: This comment seems to be wrong, I don't see how this can return `-1`.

**Comment by alvaroaleman** [comment]
@xmudrii Updated the ordering of the imports

File: pkg/controller/machineset/status.go [hunk 10]
```diff
@@ -132,3 +133,51 @@ func (c *ReconcileMachineSet) getMachineNode(machine *v1alpha1.Machine) (*corev1
...
+
+// isNodeAvailable returns true if the node is ready and minReadySeconds have elapsed or is 0. False otherwise.
+func isNodeAvailable(node *corev1.Node, minReadySeconds int32, now metav1.Time) bool {
+	if !isNodeReady(node) {
+		return false
+	}
+
+	if minReadySeconds == 0 {
+		return true
+	}
```
[the remark this answers is not in the document]
Comment (xmudrii) [reply pkg/controller/machineset/status.go]: Oh, I just got confused, we have `isNodeReady` above, which will return `false` if it isn't ready.

[the remark this answers is not in the document]
Comment (alvaroaleman) [reply pkg/controller/machineset/status.go]: At this point the code is just copied over so I can not answer this question. If we change something about it, we should IMHO do it in a dedicated PR, so git keeps recognizing this as a rename

**Pull request merged**